Changed Placement of CloseModal()

eshwanthkartitr opened the pull request:
Hi @Abhinav-ark I have changed the closeModal() position and i was not able to accidently exit out of the edit placement can you verify it once bro

eshwanthkartitr committed:
Changed Placement of CloseModal()

Abhinav-ark commented on package.json:
```diff
@@ -15,6 +15,7 @@
     "chart.js": "^4.4.1",
     "material-icons": "^1.13.12",
     "next": "^13.5.4",
+    "placement_tracker_v2": "file:",
```
Can you comment on the necessity of this dependency and remove it if not deemed necessary?

eshwanthkartitr replied:
This was unnecessary i have removed it

Abhinav-ark commented on package-lock.json:
```diff
@@ -3289,6 +3290,10 @@
         "node": ">= 6"
       }
     },
+    "node_modules/placement_tracker_v2": {
+      "resolved": "",
+      "link": true
+    },
```
Can you comment on the necessity of this dependency and remove it if not deemed necessary?

eshwanthkartitr replied:
This was unnecessary i have removed it

Abhinav-ark reviewed (changes requested):
I am getting an error in the file, `app/dashboard/admin/student/[studentId]/editPlacement/[placementID]/page.js` 

![image](https://github.com/user-attachments/assets/8cf705c7-838f-4e08-a164-845a61a8ff81)

There might be a problem with secure local storage not getting set properly in the previous page. I'd appreciate if you look into this.

Abhinav-ark commented:
Resolves #2 .

eshwanthkartitr commented:
Hey @Abhinav-ark I have changed 
const isValidJobLocation = jobLocation.length > 0 to 
const isValidJobLocation = jobLocation && jobLocation.length > 0 ? true : false;
because the joblocation was null 
![image](https://github.com/user-attachments/assets/a9565cef-1d38-490c-a560-1a895976bb83)

also i have added a joblocationMessage Variable
const jobLocationMessage = isValidJobLocation ? jobLocation : "";
which will replace with "" if the flasg was false
![image](https://github.com/user-attachments/assets/1ae4b733-1cad-409d-87a7-bf07d1981cea)

![image](https://github.com/user-attachments/assets/addca60d-1bae-441a-9211-3e6ef6219c2b)


is this fine ?

eshwanthkartitr committed:
Rectified the unassigned joblocation bug

eshwanthkartitr commented:
Hey @Abhinav-ark @Ashrockzzz2003  is there some other change which i need to do?

Ashrockzzz2003 reviewed (changes requested):
https://github.com/user-attachments/assets/9a82afcc-bbee-4924-9e9b-ccf349ed86d4

I tried running it and the bug is still there @eshwanthkartitr 

Test it by running the backend code locally. Link to the same is available in `README.md`. Checkout #7 for test credentials. To run backend code, some env vars need to be set. Refer an issue that talks about it there as well.

Good job on customizing the catch block messages.

eshwanthkartitr committed:
added type button to add company button

eshwanthkartitr committed:
added type button to add company button

eshwanthkartitr commented:
Hi @Ashrockzzz2003 I have made the changes now 
![image](https://github.com/user-attachments/assets/1c5c0f92-c218-42e5-bc79-bb7ac533e981)
can you check now

Ashrockzzz2003 reviewed (changes requested):
That bug is now fixed. But new problem. 

The newly added company must be selected in the dropdown. That's not happening. Also, after adding a new company, if I try to select a different company from the dropdown, not happening.


Build errors. `npm run build`. It was building fine before this.
![image](https://github.com/user-attachments/assets/d911fa06-dd6b-4d40-aa39-f4b1dc5a5746)

eshwanthkartitr committed:
Fixed the three mentioned errors

eshwanthkartitr committed:
Fixed error display

eshwanthkartitr commented:
Hi @Ashrockzzz2003 I have made the fixes and `npm run build` is running without any problem and there was a bug which was caused by usestate i have resolved it now

Ashrockzzz2003 commented on app/dashboard/admin/student/new/page.js:
```diff
@@ -82,19 +82,19 @@ export default function RegisterStudent() {
-        //     "isPlaced": isPlaced,
-        //     "CGPA": CGPA
-        // });
+        console.log({
+            "studentRollNo": studentRollNo,
+            "studentEmail": studentEmail,
+            "studentName": studentName,
+            "studentPassword": hashPassword(studentPassword),
+            "studentSection": studentSection,
+            "studentGender": studentGender === "Male" ? "M" : studentGender === "Female" ? "F" : "O",
+            "studentBatch": studentBatch,
+            "studentDept": studentDept,
+            "isHigherStudies": isHigherStudies === "Yes" ? "1" : "0",
+            "isPlaced": isPlaced,
+            "CGPA": CGPA
+        });
```
comment this out. needed only during dev for test.

Ashrockzzz2003 commented on app/dashboard/admin/page.js:
```diff
@@ -68,6 +68,7 @@ export default function AdminDashboard() {
                 "Authorization": "Bearer " + secureLocalStorage.getItem("userAccess"),
             }
         }).then((res) => {
+            console.log(res);
```
comment this out. needed only for dev.

Ashrockzzz2003 commented on app/dashboard/admin/page.js:
```diff
@@ -139,10 +140,11 @@ export default function AdminDashboard() {
-                        // console.log(data["companyId"]);
-                        // console.log(data["companyName"]);
-                        alertError("Error", "Something went wrong. Please try again later.");
+                        console.log(data["companyId"]);
+                        console.log(data["companyName"]);
```
comment out

Ashrockzzz2003 commented on app/dashboard/student/editPlacement/[placementID]/page.js:
```diff
@@ -1,107 +1,149 @@
+        const placements = JSON.parse(p);
+        var placement = {};
+        
+        console.log(placements);
```
comment out

Ashrockzzz2003 commented on app/dashboard/student/editPlacement/[placementID]/page.js:
```diff
@@ -387,7 +428,11 @@ export default function NewPlacementScreen() {
                                     <Dropdown
-                                        value={companyId} onChange={(e) => setCompanyId(e.value || '')} options={companyList} optionLabel="companyName" optionValue='id'
+                                        value={companyId} onChange={(e) =>{
+                                            console.log('Company selected:', e.value);
+                                            setCompanyId(e.value);
+                                            console.log('CompanyId updated:', companyId);
```
comment out logs.

Ashrockzzz2003 commented on app/forgotPassword/page.js:
```diff
@@ -42,7 +42,7 @@ export default function ForgotPassword() {
     const emailRegex = new RegExp(/^[a-zA-Z0-9._-]+@(cb.students.amrita.edu|cb.amrita.edu)$/);
 
-    const isValidEmail = emailRegex.test(userEmail) || userEmail === "ashrockzzz2003@gmail.com" || userEmail === "hsheadone@gmail.com";
+    const isValidEmail = emailRegex.test(userEmail) || userEmail === "ashrockzzz2003@gmail.com" || userEmail === "hsheadone@gmail.com" || userEmail === "eshwanthkartitr@gmail.com";
```
care to remove this or let it be there? your choice. 😂

Ashrockzzz2003 commented on app/login/page.js:
```diff
@@ -25,7 +25,7 @@ export default function Login() {
     const emailRegex = new RegExp(/^[a-zA-Z0-9._-]+@(cb.students.amrita.edu|cb.amrita.edu)$/);
 
-    const isValidEmail = emailRegex.test(userEmail) || userEmail === "ashrockzzz2003@gmail.com" || userEmail === "hsheadone@gmail.com" || userEmail === "abhinavramki2@gmail.com";
+    const isValidEmail = emailRegex.test(userEmail) || userEmail === "ashrockzzz2003@gmail.com" || userEmail === "hsheadone@gmail.com" || userEmail === "abhinavramki2@gmail.com" || userEmail === "eshwanthkartitr@gmail.com";
```
care to remove this 😂??

eshwanthkartitr replied:
I was trying to check admin mode also with my mail i will remove this now

Ashrockzzz2003 replied:
Got it da.

Ashrockzzz2003 commented on package.json:
```diff
@@ -14,7 +14,8 @@
-    "next": "^13.5.4",
+    "next": "^15.1.3",
+    "placement_tracker_v2": "file:",
     "primereact": "^9.6.2",
```
`placement_tracker_v2`
remove if not needed.

eshwanthkartitr replied:
ok

Ashrockzzz2003 reviewed (changes requested):
All bugs seem to be fixed! Good work.

Ashrockzzz2003 committed:
Comment out logs not edited in #3.

eshwanthkartitr committed:
Removed id and placement_traccker_v2 file

eshwanthkartitr committed:
Merge branch 'v2' of https://github.com/eshwanthkartitr/placement_tracker_web into v2

eshwanthkartitr committed:
Commented out unnecessary logs

eshwanthkartitr commented:
Hi @Ashrockzzz2003 i have fixed the unnecessary display of logs and removed my mailid

Ashrockzzz2003 committed:
Update code formatting.

Ashrockzzz2003 reviewed (approved):
LGTM

Good work @eshwanthkartitr

Ashrockzzz2003 commented:
/bounty 40 @eshwanthkartitr
